common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/BlocksRemoved.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.shuffle.protocol;
+
+import com.google.common.base.Objects;
+import io.netty.buffer.ByteBuf;
+
+// Needed by ScalaDoc. See SPARK-7726
+import static org.apache.spark.network.shuffle.protocol.BlockTransferMessage.Type;
+
+/** The reply to remove blocks giving back the number of removed blocks. */
+public class BlocksRemoved extends BlockTransferMessage {
+  public final int numRemovedBlocks;
+
+  public BlocksRemoved(int numRemovedBlocks) {
+    this.numRemovedBlocks = numRemovedBlocks;
+  }
+
+  @Override
+  protected Type type() { return Type.BLOCKS_REMOVED; }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(numRemovedBlocks);
+  }
+
+  @Override
+  public String toString() {
+    return Objects.toStringHelper(this)
+      .add("numRemovedBlocks", numRemovedBlocks)
+      .toString();
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (other != null && other instanceof BlocksRemoved) {
+      BlocksRemoved o = (BlocksRemoved) other;
+      return Objects.equal(numRemovedBlocks, o.numRemovedBlocks);
```

Comment:
why do we compare 2 ints with `Objects.equal`?

Reply:
IDEs are usually good at generating `equals` and `hashCode` for java classes, maybe we can use the generated version.

Reply:
You are right (I followed the pattern used within the same package like in ExecutorShuffleInfo).
I can open a minor PR with fixing these two or I can add it this tiny change into my next PR which might be opened next week or the week after. It is about avoiding the network at fetching shuffle blocks from the block manager running on the same host, so it is just loosely related.

Which one is preferred by you?

Reply:
both are fine, this is really trivial